kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/order/OrderRepository.java

```diff
@@ -152,23 +150,21 @@ private void getOrder(final String orderID) {
         remoteData.getOrder(orderID, new Callback<Order>() {
             @Override
             public void onResponse(Order order) {
+                decrementPendingOrdersCount();
                 setCompletedOrder(order);
             }
 
             @Override
             public void onFailure(Throwable t) {
-
+                decrementPendingOrdersCount();
             }
         });
     }
 
     private void decrementPendingOrdersCount() {
         if (hasMorePendingOffers()) {
-            pendingOrdersCount.decrementAndGet();
-        } else {
-            if (paymentObserver != null) {
+            if (pendingOrdersCount.decrementAndGet() == 0 && paymentObserver != null) {
```

Comment:
as we discussed add ~ paymentObserverCount to only control adding/removing  payment observer

Reply:
👍